packages/server/src/adapters/node-http/content-type/form-data/index.ts

```diff
@@ -103,3 +109,28 @@ export { createFileUploadHandler as experimental_createFileUploadHandler } from
 export { composeUploadHandlers as experimental_composeUploadHandlers } from './uploadHandler';
 export { type UploadHandler } from './uploadHandler';
 export { isMultipartFormDataRequest as experimental_isMultipartFormDataRequest };
+
+export function experimental_createFormDataMiddleware<
+  TTRPCInstance extends ReturnType<AnyTRPCInstance>,
+>(
+  t: TTRPCInstance,
+  config?: {
+    uploadHandler?: UploadHandler;
+  },
+) {
+  return t.middleware(async (opts) => {
+    // TODO: ideally we wouldn't force this to be added by the user, would be better to have the Adapter provide this
```

Comment:
If we at compile-time could enforce that `opts.ctx.req` exists, it'd help

Reply:
I actually don't think we should use ctx for this at all. My second PR (stacked in this one) addresses this issue, I'm just trying to be bite-sized about the experiments